app/scripts/controllers/permissions/index.js

```diff
@@ -4,8 +4,9 @@ import ObservableStore from 'obs-store'
 import log from 'loglevel'
 import { CapabilitiesController as RpcCap } from 'rpc-cap'
 import { ethErrors } from 'eth-json-rpc-errors'
+import { cloneDeep } from 'lodash'
 
-import getRestrictedMethods from './restrictedMethods'
+import _getRestrictedMethods from './restrictedMethods'
```

Comment:
The underscore prefix as a convention is for "private" methods, which this doesn't seem to be considering that it is exported. If we're trying to avoid shadowing the name in the ctor, we could either rename the variable there or spread the rest of the args/opts into an opts object:

```js
  constructor (
     /* istanbul ignore next */
     {
       platform, notifyDomain, notifyAllDomains, getKeyringAccounts,
       platform, notifyDomain, notifyAllDomains,
       getKeyringAccounts, ...opts,
     } = {},
     restoredPermissions = {},
     restoredState = {}) {
     // [...]
     this._restrictedMethods =(opts. getRestrictedMethods || getRestrictedMethods)(this)
```

Or after writing all that up maybe we could call this `defaultGetRestrictedMethods` or something.

Reply:
Or maybe we should pass in the restricted methods in all cases, and drop the default? i.e. actual dependency injection

I'd rather avoid adding complexity to the code under test solely for the sake of testing. I find this "optional parameter that isn't ever actually used" pattern to be confusing. If I were to come across this in the future, I'd likely end up wasting some time investigating _why_ it was added as a parameter and whether I should pass something in.

Whereas if the restricted methods are a required parameter that is _always_ passed in, there is no additional branch, and no question as to what it's for.

Reply:
Yeah, I have to agree with you there Mark; it sucks. It's now always passed in.